NPC_Behavior.cpp

```diff
@@ -1,85 +1,104 @@
 #include "NPC_Behavior.h"
+constexpr int small_enemy{ 2 };
+constexpr int big_enemy{ 5 };
+constexpr int NPC_Wounded{ 2 };
+constexpr int NPC_DIED{ 2 };
+constexpr int NPC_could_run{ 4 };
+constexpr int not_possible_to_restore_energy{ 1 };
 
-NPC::NPC() : state_{new On_guard(state_)} {}
+NPC::NPC() : state_{ new On_guard(state_) } {}
 
-bool NPC::has_energy() { return this->energy_ > 50; }
-
-void NPC::call_energy_ok() { Fight::energy_Ok; }
+bool NPC::has_energy() { return energy_ > 50; }
 
 void NPC::On_guard() {
-  auto random{rand() % 5};
-  if (random < 2) {
-    state_->see_small_enemy();
-    return;
-  }
-  state_->see_big_enemy();
+    auto random{ rand() % 5 };
+    while (random == small_enemy || random == big_enemy) {
+        if (random == small_enemy) {
+            state_->see_small_enemy();
+            return;
+        }
+        if (random == big_enemy) {
+            state_->see_big_enemy();
+            return;
+        }
+        std::cout << "NPC On guard!" << std::endl;
+    }

```

Comment:
лучше будет так его переписать
```suggestion
    while (random != small_enemy && random != big_enemy) {
        std::cout << "NPC On guard!" << std::endl;
    }
     if (random == small_enemy) {
            state_->see_small_enemy();
            return;
        }

        state_->see_big_enemy();

```
просто в твоем цикле не понятно как будет выполнена строка 
std::cout << "NPC On guard!" << std::endl;